packages/mobile/src/services/ReactNativeLogger.ts

```diff
@@ -34,13 +42,38 @@ export default class ReactNativeLogger {
     const sanitizedError =
       error && shouldSanitizeError ? this.sanitizeError(error, valueToPurge) : error
     const errorMsg = this.getErrorMessage(sanitizedError)
-    Sentry.captureException(error, { extra: { tag, message, errorMsg, source: 'Logger.error' } })
-    console.info(`${tag} :: ${message} :: ${errorMsg}`)
+    const isNetworkError = this.networkErrors.some((networkError) =>
+      errorMsg.toLowerCase().includes(networkError)
+    )
+
+    // prevent genuine network errors from being sent to Sentry
+    if (!isNetworkError || (this.isNetworkConnected && isNetworkError)) {
+      Sentry.captureException(error, {
+        extra: {
+          tag,
+          message,
+          errorMsg,
+          source: 'Logger.error',
+          networkConnected: this.isNetworkConnected,
+        },
+      })
+    }
+    console.info(
+      `${tag} :: ${message} :: ${errorMsg} :: network connected ${this.isNetworkConnected}`
+    )
     if (__DEV__) {
       console.info(console.trace())
     }
   }
 
+  setIsNetworkConnected = (isConnected: boolean) => {
+    this.isNetworkConnected = isConnected
+  }
+
+  setNetworkErrors = (errors: string[]) => {
+    this.networkErrors = [...DEFAULT_SENTRY_NETWORK_ERRORS, ...errors]
```

Comment:
The `set` naming makes me think it'd override everything instead of appending. I think it makes sense to override anyways? Unless I'm missing something

Reply:
oh yeah, this is true. if we don't have any defaults in the code and rely only on remote config values, we will have an edge case first time the user opens the app + they are offline and couldn't fetch remote config....but as i'm writing this i realise we can just set the default value of the remote config value in the code and do a redux migration. i'll go with that!

Reply:
hum. i removed the appending so this setter now just sets :) i thought i would get rid of the config variable altogether but i ended up keeping it. this logger is initialised way before the appInit saga where this setter is first called, so it makes sense to init this Logger with a default value to avoid sending network errors on app start. so now the default value is being use here and the remote config defaults, so i thought it's better to keep the source of truth in the config file